molgenis-data-rest/src/main/java/org/molgenis/data/rest/service/RestService.java

```diff
@@ -0,0 +1,153 @@
+package org.molgenis.data.rest.service;
+
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.MolgenisFieldTypes.FieldTypeEnum.CATEGORICAL;
+import static org.molgenis.MolgenisFieldTypes.FieldTypeEnum.CATEGORICAL_MREF;
+import static org.molgenis.MolgenisFieldTypes.FieldTypeEnum.MREF;
+import static org.molgenis.MolgenisFieldTypes.FieldTypeEnum.XREF;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.molgenis.MolgenisFieldTypes.FieldTypeEnum;
+import org.molgenis.data.AttributeMetaData;
+import org.molgenis.data.DataConverter;
+import org.molgenis.data.DataService;
+import org.molgenis.data.Entity;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.IdGenerator;
+import org.molgenis.data.MolgenisDataException;
+import org.molgenis.data.support.MapEntity;
+import org.molgenis.fieldtypes.BoolField;
+import org.molgenis.file.FileDownloadController;
+import org.molgenis.file.FileMeta;
+import org.molgenis.file.FileStore;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import com.google.common.collect.Lists;
+
+@Service
+public class RestService
+{
+	private final DataService dataService;
+	private final IdGenerator idGenerator;
+	private final FileStore fileStore;
+
+	@Autowired
+	public RestService(DataService dataService, IdGenerator idGenerator, FileStore fileStore)
+	{
+		this.dataService = requireNonNull(dataService);
+		this.idGenerator = requireNonNull(idGenerator);
+		this.fileStore = requireNonNull(fileStore);
+	}
+
+	/**
+	 * Creates a new MapEntity based from a HttpServletRequest
+	 * 
+	 * @param meta
+	 * @param request
+	 * @return
+	 */
+	public Entity toEntity(final EntityMetaData meta, final Map<String, Object> request)
```

Comment:
Would this also work as a spring data converter thingy? That you register it somewhere as a converter from Request to entity and that it then allows you to rewrite the REST method as one that accepts Entity parameter requests rather than HttpRequest